unstract/sdk1/src/unstract/sdk1/llm.py

```
@@ -480,6 +482,67 @@ def _record_usage(
             kwargs={"provider": self.adapter.get_provider(), **self.platform_kwargs},
         )
 
+    def _raise_for_empty_response(self, finish_reason: str | None) -> "NoReturn":
```

Ruff (0.15.6)

[error] Undefined name `NoReturn`

(F821)